lockfile_unix.go

```diff
@@ -14,17 +14,13 @@ import (
 )
 
 func getLockFile(path string, ro bool) (Locker, error) {
-	var fd int
-	var err error
-	if ro {
-		fd, err = unix.Open(path, os.O_RDONLY, 0)
-	} else {
-		fd, err = unix.Open(path, os.O_RDWR|os.O_CREATE, unix.S_IRUSR|unix.S_IWUSR)
-	}
+	// Create and open the lock file to catch potential permission errors
```

Comment:
Nothing changes as far as handling of R/O Locks?

Reply:
Nothing changes. We already have an internal counter that is incremented for each (successful) lock and decremented for each unlock. In this change, we make use of that by opening and closing the file when the counter == 0 during lock and unlock.